PWGHF/TableProducer/candidateCreator2Prong.cxx

```diff
@@ -210,14 +229,147 @@ struct HfCandidateCreator2Prong {
+      } else {
+        hDcaXYProngs->Fill(track1.pt(), -999.);
+        hDcaZProngs->Fill(track1.pt(), -999.);
+      }
+
+      KFParticle kfCandD0;
+      const KFParticle* kfDaughtersD0[2] = {&kfPosPion, &kfNegKaon};
+      kfCandD0.SetConstructMethod(2);
+      kfCandD0.Construct(kfDaughtersD0, 2);
+      KFParticle kfCandD0bar;
+      const KFParticle* kfDaughtersD0bar[2] = {&kfNegPion, &kfPosKaon};
+      kfCandD0bar.SetConstructMethod(2);
+      kfCandD0bar.Construct(kfDaughtersD0bar, 2);
+
+      auto massD0 = kfCandD0.GetMass();
+      auto massD0bar = kfCandD0bar.GetMass();
+
+      hCovSVXX->Fill(kfCandD0.Covariance(0, 0));
+      hCovSVYY->Fill(kfCandD0.Covariance(1, 1));
+      hCovSVXZ->Fill(kfCandD0.Covariance(2, 0));
+      hCovSVZZ->Fill(kfCandD0.Covariance(2, 2));
+      auto covMatrixSV = kfCandD0.CovarianceMatrix();
+
+      double phi, theta;
+      getPointDirection(std::array{KFPV.GetX(), KFPV.GetY(), KFPV.GetZ()}, array{kfCandD0.GetX(), kfCandD0.GetY(), kfCandD0.GetZ()}, phi, theta);
+      auto errorDecayLength = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, theta) + getRotatedCovMatrixXX(covMatrixSV, phi, theta));
+      auto errorDecayLengthXY = std::sqrt(getRotatedCovMatrixXX(covMatrixPV, phi, 0.) + getRotatedCovMatrixXX(covMatrixSV, phi, 0.));
+
+      float topolChi2PerNdfD0 = -999.;
+      KFParticle kfCandD0Topol2PV;
+      if (constrainKfToPv) {
+        kfCandD0Topol2PV = kfCandD0;
+        kfCandD0Topol2PV.SetProductionVertex(KFPV);
+        topolChi2PerNdfD0 = kfCandD0Topol2PV.GetChi2() / kfCandD0Topol2PV.GetNDF();
+      }
+
+      // fill candidate table rows
+      rowCandidateBase(collision.globalIndex(),
+                       KFPV.GetX(), KFPV.GetY(), KFPV.GetZ(),
+                       kfCandD0.GetX(), kfCandD0.GetY(), kfCandD0.GetZ(),
+                       errorDecayLength, errorDecayLengthXY,   // TODO: much different from the DCAFitterN one
+                       kfCandD0.GetChi2() / kfCandD0.GetNDF(), // TODO: to make sure it should be chi2 only or chi2/ndf, much different from the DCAFitterN one
+                       kfPosPion.GetPx(), kfPosPion.GetPy(), kfPosPion.GetPz(),
+                       kfNegKaon.GetPx(), kfNegKaon.GetPy(), kfNegKaon.GetPz(),
+                       impactParameter0XY, impactParameter1XY,
+                       errImpactParameter0XY, errImpactParameter1XY,
+                       rowTrackIndexProng2.prong0Id(), rowTrackIndexProng2.prong1Id(),
+                       rowTrackIndexProng2.hfflag());
+      rowCandidateKF(topolChi2PerNdfD0,
+                     massD0, massD0bar);
+
+      // fill histograms
+      if (fillHistograms) {
+        hMass2->Fill(massD0);
+        hMass2->Fill(massD0bar);
+        if (constrainKfToPv) {
+          float massD0Topol = 0., errMassD0Topol = 0.;
+          if (kfCandD0Topol2PV.GetMass(massD0Topol, errMassD0Topol))
+            hKFMassFailed->Fill(massD0); // Set up D0 topological constraints to PV failed
```

Comment:
Add braces.

Reply:
What does the comment mean?

Reply:
I removed these two histograms related to the masses after topological constraints. We agreed not to use them afterwards.